### ISSUE TITLE
Duplicate IterVar cause lower failure

### ISSUE THREAD

**Issue by wweic** [dmlc/tvm#573]
````bash
In [1]: %paste
import tvm
n = tvm.var('n')
m = tvm.var('m')
A = tvm.placeholder((m, n), name='A')
B = tvm.compute((m, n), lambda i, j: A[i, j], name='B')

s = tvm.create_schedule(B.op)
xo, yo, xi, yi = s[B].tile(B.op.axis[0], B.op.axis[1], x_factor=10, y_factor=5)

s[B].reorder(xi, yo, xi)

## -- End pasted text --

In [2]: print(tvm.lower(s, [A, B], simple_mode=True))
[18:10:06] /Users/weichen/workspace/deep-learning/tvm/dmlc-core/include/dmlc/logging.h:308: [18:10:06] src/schedule/message_passing.cc:107: Check failed: allow_missing

Stack trace returned 10 entries:
[bt] (0) 0   libtvm.dylib                        0x0000000111650e28 _ZN4dmlc15LogMessageFatalD2Ev + 40
[bt] (1) 1   libtvm.dylib                        0x000000011181035d _ZN3tvm8schedule11PassUpIndexERKNS_5StageERKNS_3MapINS_7IterVarENS_5RangeEvvEEPNSt3__113unordered_mapIS5_N6Halide4ExprENSA_4hashIS5_EENSA_8equal_toIS5_EENSA_9allocatorINSA_4pairIKS5_SD_EEEEEEb + 3437
[bt] (2) 2   libtvm.dylib                        0x0000000111726157 _ZN3tvm2op12MakeLoopNestERKNS_5StageERKNSt3__113unordered_mapINS_7IterVarENS_5RangeENS4_4hashIS6_EENS4_8equal_toIS6_EENS4_9allocatorINS4_4pairIKS6_S7_EEEEEEmbRKNS4_13unordered_setIS6_S9_SB_NSC_IS6_EEEEPNS5_IS6_N6Halide4ExprES9_SB_NSC_INSD_ISE_SQ_EEEEEE + 6583
[bt] (3) 3   libtvm.dylib                        0x000000011171916d _ZN3tvm15ComputeLoopNest4makeEPKNS_13ComputeOpNodeERKNS_5StageERKNSt3__113unordered_mapINS_7IterVarENS_5RangeENS7_4hashIS9_EENS7_8equal_toIS9_EENS7_9allocatorINS7_4pairIKS9_SA_EEEEEE + 413
[bt] (4) 4   libtvm.dylib                        0x0000000111718439 _ZN3tvm15MakeComputeStmtEPKNS_13ComputeOpNodeERKNS_5StageERKNSt3__113unordered_mapINS_7IterVarENS_5RangeENS6_4hashIS8_EENS6_8equal_toIS8_EENS6_9allocatorINS6_4pairIKS8_S9_EEEEEE + 41
[bt] (5) 5   libtvm.dylib                        0x000000011171a726 _ZNK3tvm13ComputeOpNode12BuildProvideERKNS_5StageERKNSt3__113unordered_mapINS_7IterVarENS_5RangeENS4_4hashIS6_EENS4_8equal_toIS6_EENS4_9allocatorINS4_4pairIKS6_S7_EEEEEE + 150
[bt] (6) 6   libtvm.dylib                        0x00000001118326ba _ZN3tvm8schedule12MakePipelineERKNS_5StageERKNSt3__113unordered_mapINS_7IterVarENS_5RangeENS4_4hashIS6_EENS4_8equal_toIS6_EENS4_9allocatorINS4_4pairIKS6_S7_EEEEEEN6Halide8Internal4StmtE + 58
[bt] (7) 7   libtvm.dylib                        0x000000011183347a _ZN3tvm8schedule11ScheduleOpsENS_8ScheduleENS_3MapINS_7IterVarENS_5RangeEvvEE + 2010
[bt] (8) 8   libtvm.dylib                        0x0000000111686057 _ZNSt3__110__function6__funcIN3tvm8schedule3$_8ENS_9allocatorIS4_EEFvNS2_7runtime7TVMArgsEPNS7_11TVMRetValueEEEclEOS8_OSA_ + 183
[bt] (9) 9   libtvm.dylib                        0x00000001119a2f7b TVMFuncCall + 75
````

I think duplicate IterVar is never expected, right? I would like to check during `reorder` and print meaningful error messages like `Same axis can not appear more than once`, does this make sense?

**Comment by tqchen** [comment]
Yes, this is definitely a better error message

**Comment by wweic** [comment]
I have a patch on python side. But maybe it's better to alert on C++ side? So all language bindings will have this check automatically.

**Comment by tqchen** [comment]
fixed by #575
